core/embed/extmod/trezorobj.h

```diff
@@ -75,6 +75,25 @@ static inline uint8_t trezor_obj_get_uint8(mp_obj_t obj) {
   return u;
 }
 
+static inline uint64_t trezor_obj_get_uint64(mp_const_obj_t obj) {
+  if (obj == mp_const_false) {
+    return 0;
+  } else if (obj == mp_const_true) {
+    return 1;
+  } else if (MP_OBJ_IS_SMALL_INT(obj)) {
+    mp_int_t i = MP_OBJ_SMALL_INT_VALUE(obj);
+    mp_uint_t u = i;
+    return u;
+  } else if (MP_OBJ_IS_TYPE(obj, &mp_type_int)) {
+    uint64_t u = 0;
+    mp_obj_int_t *self = MP_OBJ_TO_PTR(obj);
+    mpz_as_bytes(&self->mpz, false, 8, (byte *)&u);
```

Comment:
I'm not sure if it's important:

  * Since `mpz_as_bytes` [accepts](https://github.com/micropython/micropython/blob/97a7cc243b028833bdcb8ce0bc19b2bce7545851/py/mpz.c#L1603) negative numbers, `trezor_obj_get_uint64` accepts them too (whereas `trezor_obj_get_uint` does not).
  * I guess `trezor_obj_get_uint64` is not endianness agnostic.

Reply:
The `false` argument to `mpz_as_bytes` says this is all little-endian.

Reply:
> The false argument to mpz_as_bytes says this is all little-endian.

Which means it won't work on big-endian architecture. (Honestly, I'm not sure whether we want to support such architectures).

Reply:
Fixed endianness in 4f896bd256e2289af073df37835e8da80a9c5eb4